linker/STM32F4xx/STM32F446RE.ld

```diff
@@ -0,0 +1,8 @@
+MEMORY {
```

Comment:
We don't need this file. You can remove this one.

The make file will include the linker files based on the device name you configure in your project's make file. So in your case you should define it as STM32F446xE and then it will include the STM32F446xE.ld linker script. The R is not relevant since it defines the number of pins the device has.